components/payments/content/bat_payment_app_factory.cc

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/payments/content/bat_payment_app_factory.h"
+
+#include <utility>
+
+#include "brave/components/brave_rewards/common/constants.h"
+#include "brave/components/payments/content/bat_payment_app.h"
+#include "components/payments/content/content_payment_request_delegate.h"
+#include "components/payments/content/payment_request.h"
+#include "components/payments/content/payment_request_spec.h"
+
+namespace payments {
+
+BatPaymentAppFactory::BatPaymentAppFactory()
+    : PaymentAppFactory(PaymentApp::Type::INTERNAL) {}
+
+BatPaymentAppFactory::~BatPaymentAppFactory() = default;
+
+// static
+bool BatPaymentAppFactory::IsBatSupportedMethod(
+    base::WeakPtr<PaymentRequestSpec> spec) {
+#if defined(OS_ANDROID)
```

Comment:
this should be using the buildflag, not OS_ANDROID

Reply:
actually better yet we should just not insert this code at all if the buildflag is false

Reply:
resolved